heat/core/logical.py

```diff
@@ -516,17 +516,14 @@ def sanitize_input_type(
     x = sanitize_input_type(x, y)
     y = sanitize_input_type(y, x)
 
-    # if one of the tensors is distributed, unsplit/gather it
-    if x.split is not None and y.split is None:
-        t1 = manipulations.resplit(x, axis=None)
-        return t1, y
+    # Assign t1 and t2 with x and y, with t1 potentially representing the distributed operand (if x is distributed)
+    t1, t2 = x, y
 
-    elif x.split != y.split:
-        t2 = manipulations.resplit(y, axis=x.split)
-        return x, t2
+    #  If y is distributed, update t1 and t2 accordingly
+    if y.split is not None:
+        t1, t2 = y, x
 
-    else:
-        return x, y
+    return t1, t2
```

Comment:
The output should be distributed along the same split axis, but now the inputs are simply returned. This is not what we want here.